src/Api/Vault/Controllers/CiphersController.cs

```diff
@@ -1120,19 +1130,25 @@ public async Task<CipherMiniResponseModel> PutUnarchive(Guid id)
             throw new BadRequestException("Cipher was not unarchived. Ensure the provided ID is correct and you have permission to archive it.");
         }
 
-        return new CipherMiniResponseModel(unarchivedCipherDetails.First(), _globalSettings, unarchivedCipherDetails.First().OrganizationUseTotp);
+        return new CipherResponseModel(unarchivedCipherDetails.First(),
+            await _userService.GetUserByIdAsync(userId),
+            await _applicationCacheService.GetOrganizationAbilitiesAsync(),
+            _globalSettings
+        );
     }
 
     [HttpPut("unarchive")]
     [RequireFeature(FeatureFlagKeys.ArchiveVaultItems)]
-    public async Task<ListResponseModel<CipherMiniResponseModel>> PutUnarchiveMany([FromBody] CipherBulkUnarchiveRequestModel model)
+    public async Task<ListResponseModel<CipherResponseModel>> PutUnarchiveMany([FromBody] CipherBulkUnarchiveRequestModel model)
     {
         if (!_globalSettings.SelfHosted && model.Ids.Count() > 500)
         {
             throw new BadRequestException("You can only unarchive up to 500 items at a time.");
         }
 
         var userId = _userService.GetProperUserId(User).Value;
+        var user = await _userService.GetUserByIdAsync(userId);
```

Comment:
🎨 We should use `GetUserByPrincipalAsync` to avoid an extra DB request. (see the other controller endpoints for examples)